go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/filecoin-project/go-fil-commp-hashhash
 
-go 1.11
+go 1.15
```

Comment:
@marten-seemann I see this thing is alive now. Question: if I *do* want to support old go-versions , could I opt out ( merge just parts of this ) ? Or will it break the integration?

Reply:
Sorry for the late reply.

It will break the integration, in the sense that every time *anything* is updated in `protocol/.github`, it will open a new PR here (no new PR is created if there's no diff between the files in `protocol/.github` and here).

Why do you want to support old Go versions? Setting the version to 1.15 in go.mod doesn't mean that the code won't compile on old Go versions btw, it just means that *if* compilation fails on an old version, it will tell you that it's likely due to the Go version you're using. Is that an acceptable behavior?

Reply:
@marten-seemann @mvdan @galargh did the above ( ability to test older versions ) get a resolution, or that's not something the tooling will support? It is not a showstopper, but it would be nice to freeze this module at v1.11 if possible.

Reply:
It's been some time so perhaps I'm lacking context, but I think Marten's last question is still valid. Bumping this to `go 1.15` or later doesn't explicitly mean that you're breaking older Go versions - at the Go code language level, it's just a signal, not a hard minimum.

Reply:
FYI, it's being upped to v1.16 now.

Reply:
@mvdan oh, I was hoping the CI would run against the minversion, so i will catch me doing something dumb that I wouldn't notice otherwise ( exactly like my unwittingly breaking i386 compilation )

Reply:
If supporting ancient Go versions is a requirement here (I still don't understand why), you could add a separate workflow that uses that version. We don't have any other repository that's interested in supporting Go versions that are outside of the Go team's maintenance window, so this is something that probably won't get added to Unified CI any time soon.

Reply:
I think this issue got a bit into the weeds. Let's backtrack:

- At T0 The repository has a `go 1.11` stanza, which I placed here for a reason, and would like to keep. It is a signal that "yes, I test this under ancient versions manually, dear user, please us your ancient golang without fear"
- A new workflow is being added to the repository which changes this from under me
- The "oh, can't the wokflow just test older versions on demand?" is a distraction - let's ignore this request
- At this stage all I would like to do is undo the automatic change to `go.mod` / backstop the `go 1.11` meta-label

If the above is impossible-on-pinciple for repositories under `github.com/filecoin-project`, this is fine albeit annoying: I can always make a fork of said repo when I get back to the larger project where having accurate `go.mod`'s is a concern.

In short: is there a way to exempt a project from `go.mod` being edited?

Reply:
> In short: is there a way to exempt a project from go.mod being edited?

Currently there’s not, and it might not be easy to build. Before investing a lot of time in building this feature, I’d like to understand why we would care that much about Go 1.11 users (especially since large parts of our stack definitely won’t even build using Go 1.15 or older). What value does this add?

I’d also like to add that just having `go 1.11` doesn’t necessarily mean that your project will still build using Go 1.11. If you actually want to ensure that, there’s probably no way around adding a test (or at least a compilation step) that actually uses Go 1.11.

The Go team only releases updates (incl. security fixes!) for the 2 most recent Go versions, so IMHO there’s no excuse for using any Go version older than that, especially if you’re building software that’s connected to the internet.

Reply:
> you actually want to ensure that, there’s probably no way around adding a test (or at least a compilation step) that actually uses Go 1.11.

This is precisely what I did locally when I built this library. This is also what I inquired being possible earlier in this thread ( answer is "not easy/possible", and that's ok )

>  I’d like to understand why we would care that much about Go 1.11 users

This lib was extracted from a stack, which among some of its requirements has "use the ancient golang that came with your ancient distro to compile and experiment". You might notice the absence from `go.mod` of all the usual suspects as well, precisely so that "large parts of our stack definitely won’t even build using Go 1.15 or older" won't apply.

> and it might not be easy to build

This is sufficient information to resolve this 🧵, as I have workarounds/way forward. I had to ask because one never knows if there's a one-line change that would keep everyone happy. Let's merge this PR when the rest of the `-short` discussion is resolved.

Thanks!